exporter/stackdriverexporter/stackdriver.go

```diff
@@ -115,20 +116,26 @@ func (se *stackdriverExporter) pushMetricsData(ctx context.Context, md consumerd
 	return se.exporter.PushMetricsProto(ctx, md.Node, md.Resource, md.Metrics)
 }
 
-// TODO(songya): add an interface PushSpanProto to Stackdriver exporter and remove this method
-// pushTraceData is a wrapper method on StackdriverExporter.PushSpans
+// pushTraceData is a wrapper method on StackdriverExporter.PushTraceSpans
 func (se *stackdriverExporter) pushTraceData(ctx context.Context, td consumerdata.TraceData) (int, error) {
 	var errs []error
 	goodSpans := 0
+	spans := make([]*trace.SpanData, 0, len(td.Spans))
+
 	for _, span := range td.Spans {
 		sd, err := spandatatranslator.ProtoSpanToOCSpanData(span)
 		if err == nil {
-			se.exporter.ExportSpan(sd)
+			spans = append(spans, sd)
 			goodSpans++
 		} else {
 			errs = append(errs, err)
 		}
 	}
 
+	err := se.exporter.PushTraceSpans(ctx, td.Node, td.Resource, spans)
+	if err != nil {
+		errs = append(errs, err)
```

Comment:
Do we consider `goodSpans` to be a valid measure if PushTraceSpans failed?

Reply:
PushTraceSpans should return droppedSpans consistent with PushMetricsProto even though there is no condition under which it drops any span right now.

Reply:
opened https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/issues/250

Reply:
I see what you're saying re: goodSpans, I think it should return 0 if PushTraceSpans has an error